File: WordCounterJava/FileHandler.java
import java.io.*;
import java.nio.file.*;
import java.util.*;

public class FileHandler {
    
    public static String readFile(String filename) throws IOException {
        Path filePath = Paths.get(filename);
        
        if (!Files.exists(filePath)) {
            throw new FileNotFoundException("File '" + filename + "' not found.");
        }
        
        long fileSize = Files.size(filePath);
        long maxSize = 5 * 1024 * 1024;
        
        if (fileSize > maxSize) {
            double fileSizeMB = fileSize / (1024.0 * 1024.0);
            throw new IOException(String.format("File size (%.2f MB) exceeds 5 MB limit.", fileSizeMB));
        }
        
        return new String(Files.readAllBytes(filePath), "UTF-8");
    }
    
    public static Set<String> loadStopwords(String stopwordsFile) {
        try {
            String content = readFile(stopwordsFile);
            Set<String> stopwords = new HashSet<>();
            
            String[] lines = content.split("\n");
            for (String line : lines) {
                String word = line.trim().toLowerCase();
                if (!word.isEmpty()) {
                    stopwords.add(word);
                }
            }
            
            return stopwords;
        } catch (IOException e) {
            System.out.println("Warning: Stopwords file '" + stopwordsFile + "' not found. Proceeding without stopwords.");
            return new HashSet<>();
        }
    }
    
    public static void exportResults(List<Map.Entry<String, Integer>> sortedCounts, String outputFile) {
        try (PrintWriter writer = new PrintWriter(new FileWriter(outputFile, false))) {
            for (Map.Entry<String, Integer> entry : sortedCounts) {
                writer.println(entry.getKey() + ": " + entry.getValue());
            }
            System.out.println("\nResults exported to '" + outputFile + "'");
        } catch (IOException e) {
            System.out.println("Error writing to file '" + outputFile + "': " + e.getMessage());
        }
    }
}

File: WordCounterPython/file_handler.py
import os


def read_file(filename):
    if not os.path.exists(filename):
        raise FileNotFoundError(f"File '{filename}' not found.")
    
    file_size = os.path.getsize(filename)
    if file_size > 5 * 1024 * 1024:
        raise ValueError(f"File size ({file_size / (1024*1024):.2f} MB) exceeds 5 MB limit.")
    
    with open(filename, 'r', encoding='utf-8') as file:
        content = file.read()
    
    return content


def load_stopwords(stopwords_file):
    try:
        content = read_file(stopwords_file)
        stopwords = set(word.strip().lower() for word in content.split('\n') if word.strip())
        return stopwords
    except FileNotFoundError:
        print(f"Warning: Stopwords file '{stopwords_file}' not found. Proceeding without stopwords.")
        return set()


def export_results(sorted_counts, output_file):
    try:
        with open(output_file, 'w', encoding='utf-8') as file:
            for word, count in sorted_counts:
                file.write(f"{word}: {count}\n")
        print(f"\nResults exported to '{output_file}'")
    except IOError as e:
        print(f"Error writing to file '{output_file}': {e}")